README.md

```diff
@@ -1,8 +1,42 @@
-# php-cpu-core-detection
+# PHP CPU Core Detection
+
+This PHP script detects the number of CPU cores available on a Linux server. It can be helpful for optimizing performance in multi-threaded applications by dynamically assessing hardware capabilities.
+
+## Purpose
+
+This tool enables developers to retrieve the number of CPU cores on a Linux server using PHP. This is especially useful for performance optimizations and load balancing in server environments.
+
+## Features
+
+- Detects the number of CPU cores on Linux-based systems.
+- Utilizes shell commands (`hash`, `nproc`, `taskset`) to gather system information.
+- Lightweight and easy to integrate into existing PHP applications.
 
 ## Supported OS/Programs
 
-Currently only Linux with `hash`, `nproc` and `taskset` are supported.
+Currently supports **Linux** only, using the following utilities:
+- `hash`
+- `nproc`
+- `taskset`
+
+## Clone this repository(Installation):
+   ```bash
+git clone https://github.com/your-repo/php-cpu-core-detection.git
+   ```
+
+## Contributing
+Contributions are welcome! To contribute:
+
+1. Fork this repository.
+2. Make your changes in a new branch.
+3. Submit a pull request with a detailed description of your changes.
+
+## Examples
+
+Check out these examples to see the CPU core detection in action:
+
+- [Asynchronous Example](examples/async.php)
+- [Synchronous example](examples/sync.php)
```

Comment:
Can you put the core (everything except use statements and inclusion of the autoloader) of the example with a link to it instead of just the link?

Reply:
idk whats the core of example i made up readme file on guess seems like its fine like this i dk anything about php thats why im adding it like that sorry but can u do that from your side rest i did from my end @WyriHaximus check it sir

Reply:
@Anuraj-IND Well, you decided to contribute to a repository for a language you don't really know, which isn't ideal, but I can work with that. Knowing the language isn't perse a requirement, but would have been helpful. What will help you further down is recognizing different techniques/patterns regardless of the language. In PHP's case these lines `"include"` which classes/functions we use: https://github.com/WyriHaximus/php-cpu-core-detector/blob/master/examples/async.php#L3-L5 while this one loads the autoloader: https://github.com/WyriHaximus/php-cpu-core-detector/blob/master/examples/async.php#L7

So for this package the core of the examples would be https://github.com/WyriHaximus/php-cpu-core-detector/blob/master/examples/async.php#L9-L20 and https://github.com/WyriHaximus/php-cpu-core-detector/blob/master/examples/sync.php#L8-L15

Reply:
done @WyriHaximus thanks again because of you i was able to learn something new in markdown i.e. how to link properly thanks i merged it in my pull request